### ISSUE TITLE
Rule proposal: no-unnecessary-type-arguments-ignore-alias

### ISSUE THREAD

**Issue by kklocker** [typescript-eslint/typescript-eslint#4632]
### Description

It would be nice to have the possibility of ignoring explicitly typed arguments (aliases) in `no-unnecessary-type-arguments`.
I am working on a codebase where the pattern `type MyString = string;` is frequently used. These are passed to generics which takes `string` as default arguments. Using `no-unnecessary-type-arguments` gives an error -- expected behavior. It would be nice to have a toggle to specify if aliases should be ignored or not

Extending #4618 since it was closed due to expected behavior. (code samples also copied from there).

### Fail

```ts
type MyTest<T extends string = string> = { value: T };
const shouldGiveError: MyTest<string> = { value: "test" };
```

Should give error on `MyTest<string>`

### Pass

```ts
type SpecialStringType = string;
type MyTest<T extends string = string> = { value: T };
const shouldNotGiveError: MyTest<SpecialStringType> = { value: "test" };
```

**Comment by bradzacher** [comment]
As mentioned in #4618 - the types are exactly equivalent.
When we ask TS for the two types and compare them - they are exactly the same type.
This is all we see.

If you do this:
```ts
type MySpecialString = string;
type Test<T = MySpecialString> = T;
type Example1 = Test<MySpecialString>;
```

When we inspect the types it will look exactly the same as either of these cases:

```ts
type Example2 = Test<string>;

type Foo = string;
type Example3 = Test<Foo>;

type Test2<T = string> = T;
type Example4 = Test2<string>;
type Example5 = Test2<MySpecialString>;
type Example6 = Test2<Foo>;
```

Aliases for primitive types are transparent to the type system, which means they are transparent to us as well as the rule is built on top of the type system.

This is why the original issue is working as intended and was closed. There's no functionality to build on top of to make this work.

---

However, if instead of a pure primitive type-alias, you created a pseudo-nominal type (also called a "branded primitive"), then that would work out of the box with no changes because it creates a new type in the type system.

```ts
type MyBrandedStringType = string & { __brand: unknown };
type Example7 = Test<MySpecialStringType>;
```

This approach has the benefit it uniquely types the strings - so you cannot just assign any old `string` to it.
EG we use this pattern in our repo so we can ensure that certain places specifically use validated "canonical paths" instead of just any old path string.

https://github.com/typescript-eslint/typescript-eslint/blob/a02c63a264e5a6c3557468a8eff48d34ca2b718a/packages/typescript-estree/src/create-program/shared.ts#L48
https://github.com/typescript-eslint/typescript-eslint/blob/a02c63a264e5a6c3557468a8eff48d34ca2b718a/packages/typescript-estree/src/create-program/shared.ts#L57-L63
